addons/mail/static/src/core/common/thread_model.js

```diff
@@ -138,6 +138,19 @@ export class Thread extends Record {
             return this.computeCorrespondent();
         },
     });
+    correspondentCountry = Record.one("Country", {
+        /** @this {import("models").Thread} */
+        compute() {
+            return this.correspondent?.persona?.country ?? this.anonymous_country;
```

Comment:
Do we really have scenarios where a channel member is not linked to any persona? That's a bit concerning :sweat:

Reply:
This is a potential point of failure. I put the optional guard for safety.
Without it, we must ensure all fetching of channel member necessarily fetch the persona.

I understand we want to live in a perfect world where everything is necessarily fetched, but there's nothing enforcing it other than proving all flows do it. I don't trust it, JS models is not (yet?) designed to ensure this is eventually fetched, so the optional chaining it there to take that into account.

Note that JS models can insert a channel member as long as it gets an id. So this is quite easy to have code that has a channel member without local knowledge of persona. I want us to put a higher priority on robust code than "efficient" code (here 'efficient' not in the sense of performant but in the sense of code that makes heavy deduction on implementation details)

Reply:
In this case that's OK because we'll just miss the flag which is totally acceptable. However, the channel member model is really defined by channel/persona. If we were to display the persona name somewhere in the UI it would look broken, even if it didn't crash. Optional chaining can hide real issues. But as I said, OK for this compute I just wanted to know if there were known scenarios where the persona is missing :)

Reply:
@tsm-odoo Until we don't silently and automatically fetch what's considered important data (like the `channel member name`, in case we lack it), I much prefer optional chaining than slowing down technical improvements by us running being bugs here and there ^^'

Reply:
Fetching stuff silently "when we need it" might just be too smart/too heavy. We now have a single formatter (`_to_store`) which sends the persona by default. Since there is a single format point, we just need to ensure it sends the correct data and it's done. There is no point in sending a partial member, hoping for a magical black box to decide it's time to fetch the lacking values while we know we need this information since the begining.

Reply:
@tsm-odoo the "we need it there so let's fetch it earlier in all cases" puts a lot of coupling in the code of total knowledge, which results in poor performance like we have in the messaging menu or discuss sidebar with the pinned conversation.

I've been trying for quite some time to convince us to move to fetch when needed, so more an automatic reactive pattern rather than proactive manual labour on our end, but I guess I'd have to work on JS models and show how code maintenance is significantly reduced by such architecture compared to what we have now

Reply:
That's why I mentioned "at the field level". I agree that we need to be less aggressive with fetches. Performance issues arise with stuff like 1k whatsapp channels being loaded at once in the sidebar. In this case, implementing a lazy load mechanism will be significantly faster and will fix the performance issue. Much more than omitting the persona when fetching a channel member.